Polygon_mesh_processing/test/Polygon_mesh_processing/test_pmp_locate.cpp

```diff
@@ -0,0 +1,827 @@
+  double x, y;
+  while(in >> x >> y)
+    tr.insert(typename RT::Point(x, y));
+
+  std::ofstream out("triangulation.off");
+  out << "OFF\n";
+  out << tr.number_of_vertices() << " " << std::distance(tr.finite_faces_begin(), tr.finite_faces_end()) << " 0\n";
+
+  std::size_t counter = 0;
+  std::map<typename RT::Point_2, std::size_t> ids;
+  for(const auto& v : CGAL::make_range(tr.finite_vertices_begin(), tr.finite_vertices_end()))
+  {
+    out << v.point().point() << " 0\n";
+    if(ids.insert(std::make_pair(v.point().point(), counter)).second)
+      ++counter;
+  }
+
+  for(const auto& fd : CGAL::make_range(tr.finite_faces_begin(), tr.finite_faces_end()))
+  {
+    out << "3 " << ids[fd.vertex(0)->point().point()] << " " << ids[fd.vertex(1)->point().point()] << " " << ids[fd.vertex(2)->point().point()] << "\n";
+  }
+
+  out.close();
+
+  std::cout << " (" << tr.number_of_vertices() << " vertices)..." << std::endl;
+  std::cout << "Kernel: " << typeid(K()).name() << std::endl;
+
+  test_locate(tr, rnd);
+}
+
+template<typename K>
+void test_2D_surface_mesh(const char* fname, CGAL::Random& rnd)
+{
+  typedef typename K::Point_2                                 Point;
+  typedef CGAL::Surface_mesh<Point>                           Mesh;
+
+  std::cout << "Testing Surface_mesh " << fname << "..." << std::endl;
+  std::cout << "Kernel: " << typeid(K()).name() << std::endl;
+
+  std::ifstream input(fname);
+  assert(input.good());
+
+  Mesh tm;
+  if(!input || !(input >> tm))
+  {
+    std::cerr << "Error: cannot read file.";
+    return;
+  }
+
+  test_locate(tm, rnd);
+}
+
+template<typename K>
+void test_3D_surface_mesh(const char* fname, CGAL::Random& rnd)
+{
+  typedef typename K::Point_3                                 Point;
+  typedef CGAL::Surface_mesh<Point>                           Mesh;
+
+  std::cout << "Testing Surface_mesh " << fname << "..." << std::endl;
+  std::cout << "Kernel: " << typeid(K()).name() << std::endl;
+
+  std::ifstream input(fname);
+  Mesh tm;
+  if(!input || !(input >> tm))
+  {
+    std::cerr << "Error: cannot read file.";
+    return;
+  }
+
+  test_locate(tm, rnd);
+}
+
+template<typename K>
+void test_polyhedron(const char* fname, CGAL::Random& rnd)
+{
+  typedef CGAL::Polyhedron_3<K>                               Polyhedron;
+
+  std::cout << "Testing Polyhedron_3 " << fname << "..." << std::endl;
+  std::cout << "Kernel: " << typeid(K()).name() << std::endl;
+
+  std::ifstream input(fname);
+  Polyhedron poly;
+  if(!input || !(input >> poly))
+  {
+    std::cerr << "Error: cannot read file.";
+    return;
+  }
+
+  test_locate(poly, rnd);
+}
+
+template <typename K>
+void test(CGAL::Random& rnd)
+{
+  test_2D_triangulation<K>("data/stair.xy", rnd);
+//  test_2D_surface_mesh<K>("data/blobby_2D.off", rnd); // temporarily, until Surface_mesh's IO is "fixed"
+  test_3D_surface_mesh<K>("data/mech-holes-shark.off", rnd);
+  test_polyhedron<K>("data-coref/elephant_split_2.off", rnd);
+}
+
+int main()
+{
```

Comment:
To avoid problems related to 32 bits x86, add this line:
```suggestion
{
  CGAL::Set_ieee_double_precision pfr;
```

That will probably magically fix your test results.